### ISSUE TITLE
Transition from non-TLS to TLS for PeerURLs is not supported for already deployed Etcd HA cluster.

### ISSUE THREAD

**Issue by ishan16696** [gardener/etcd-druid#886]
**How to categorize this issue?**
/area scalability
/kind enhancement

**What would you like to be added**:
During the testing of various scenarios, @unmarshall and I discovered that the transition from non-TLS peerURLs to TLS-enabled peerURLs is not functioning correctly for a already deployed High Availability (HA) cluster.

Here are the steps to re-create it: 

1. Deploy a 3 member etcd cluster with no TLS enabled for peerURLs.
2. Then update the etcd custom resource to enable TLS for peerURLs in this 3-member cluster.
3. Then it can observed that this scenario is currently not supported and all member won't able to come up healhty.

Etcd cluster state will be similar to this where 1 or 2 member's peerURL have been updated correctly but not for all members.
```
+------------------+---------+-------------+-----------------------------------------------------+-----------------------------------------------------+------------+
|        ID        | STATUS  |    NAME     |                     PEER ADDRS                      |                    CLIENT ADDRS                     | IS LEARNER |
+------------------+---------+-------------+-----------------------------------------------------+-----------------------------------------------------+------------+
| 1ada1b2af131c3a6 | started | etcd-test-0 |  http://etcd-test-0.etcd-test-peer.default.svc:2380/ | https://etcd-test-0.etcd-test-peer.default.svc:2379/ |      false |
| 22751b9cefc1d3fa | started | etcd-test-1 |  http://etcd-test-1.etcd-test-peer.default.svc:2380/ | https://etcd-test-1.etcd-test-peer.default.svc:2379/ |      false |
| 38cc4b73dace1a00 | started | etcd-test-2 | https://etcd-test-2.etcd-test-peer.default.svc:2380/ | https://etcd-test-2.etcd-test-peer.default.svc:2379/ |      false |
+------------------+---------+-------------+-----------------------------------------------------+-----------------------------------------------------+------------+
```

**Root cause**:

The underlying cause of this issue is that in etcd, updating peerURLs necessitates the restart of etcd member for the changes to properly propagate throughout the cluster, as it is a cluster-wide operation. More details can be found in the etcd documentation: https://etcd.io/docs/v3.5/op-guide/runtime-configuration/#update-advertise-peer-urls

**Why is this needed**:
It's not a gardener use case as in gardener always enable the peerURLs TLS for HA cluster (replicas > 1) but it's quite possible that some community users can migrate from singleton cluster to HA cluster without initially enabling TLS for peerURLs. Upon realizing the importance of enabling TLS, these users would undoubtedly want to activate peerURL TLS for their already operational HA cluster.

/cc @unmarshall

**Comment by ishan16696** [comment]
There 2 PRs will fix this issue:
1. Backup-restore PR: https://github.com/gardener/etcd-backup-restore/pull/788 where it triggers a restart of the etcd member if etcd's advertised peerURL found to be updated.
2. Etcd-wrapper PR: https://github.com/gardener/etcd-wrapper/pull/31 a `/stop` endpoint is introduced

**Comment by ishan16696** [comment]
Both of PRs which are required to fix the issue mentioned above https://github.com/gardener/etcd-druid/issues/886#issuecomment-2393572704 have been merged and backup-restore version `v0.30.2` and etcd-wrapper version `v0.2.0` which contains these PRs fixes have been released. Hence closing this issue.
/close
